nsock/src/nsock_write.c

```diff
@@ -181,7 +181,6 @@ nsock_event_id nsock_printf(nsock_pool ms_pool, nsock_iod ms_iod,
   assert(nse);
 
   res = Vsnprintf(buf, sizeof(buf), format, ap);
-  va_end(ap);
```

Comment:
I believe that you have identified a valid defect but this is not a correct fix. Quoting from section 7.21.6.8 of ISO/IEC 9899:201x:

> As the functions vfprintf, vfscanf, vprintf, vscanf, vsnprintf, vsprintf, and vsscanf invoke the va_arg macro, the value of [ap] after the return is indeterminate.

What needs to be done instead is to wrap each invocation of `Vsnprintf` with its own pair of `va_start` and `va_end`.